src/ethereum/wallet.rs

```diff
@@ -31,45 +29,41 @@ const DERIVATION_PATH: EthereumDerivationPath<wagyu_ethereum::network::Mainnet>
 pub struct Wallet {
     private_key: clarity::PrivateKey,
     geth_client: Client,
-    dai_contract_addr: Address,
-    pub chain_id: ChainId,
+    chain: ethereum::Chain,
 }
 
 impl Wallet {
-    pub async fn new(
-        seed: Seed,
-        url: Url,
-        dai_contract_addr: Address,
-        chain_id: ChainId,
-    ) -> anyhow::Result<Self> {
+    pub async fn new(seed: Seed, url: Url, chain: ethereum::Chain) -> anyhow::Result<Self> {
         let geth_client = Client::new(url);
 
         let private_key = Self::private_key_from_seed(&seed)?;
         let wallet = Self {
             geth_client,
-            dai_contract_addr,
-            chain_id,
             private_key,
+            chain,
         };
 
-        wallet.assert_chain(chain_id).await?;
+        wallet.assert_chain(chain.chain_id()).await?;
 
         Ok(wallet)
     }
+
     #[cfg(test)]
     pub fn new_from_private_key(
         private_key: clarity::PrivateKey,
         url: Url,
         chain_id: ChainId,
     ) -> Self {
         let geth_client = Client::new(url);
-        let dai_contract_adr = Address::random();
 
+        // In tests, the DAI contract doesn't exist until after _we_
+        // deploy it. We will replace this value once that happens
+        let random_dai_contract_address = Address::random();
```

Comment:
```suggestion
        let random_dai_contract_address = Address::default();
```
Setting a placeholder to `0x0..0` makes more sense to me.

Reply:
Oh, nice point!